bot/exts/evergreen/issues.py

```diff
@@ -159,79 +197,74 @@ async def issue(
-        elif isinstance(result, str):
-            await ctx.send(result)
+        results = [await self.fetch_issues(number, repository, user) for number in numbers]
+        await ctx.send(embed=self.format_embed(results, user, repository))
 
     @commands.Cog.listener()
     async def on_message(self, message: discord.Message) -> None:
         """Command to retrieve issue(s) from a GitHub repository using automatic linking if matching <repo>#<issue>."""
-        # Ignore messages not in whitelisted categories / channels, only when in guild.
-        if message.guild and not (
-            message.channel.category.id in WHITELISTED_CATEGORIES
-            or message.channel.id in WHITELISTED_CHANNELS_ON_MESSAGE
-        ):
+        if not self.repo_regex:
+            log.warning("repo_regex isn't ready, cannot look for issues.")
+            return
+
+        # Ignore bots
+        if message.author.bot:
             return
 
-        message_repo_issue_map = re.findall(fr"({self.repo_regex})#(\d+)", message.content)
+        # `issues` will hold a list of two element tuples `(repository, issue_number)`
+        issues = re.findall(fr"({self.repo_regex})#(\d+)", message.content)
         links = []
 
-        if message_repo_issue_map:
+        if issues:
+            # Block this from working in DMs
             if not message.guild:
                 await message.channel.send(
                     embed=discord.Embed(
                         title=random.choice(NEGATIVE_REPLIES),
                         description=(
-                            "You can't retreive issues from DMs. "
+                            "You can't retrieve issues from DMs. "
                             f"Try again in <#{Channels.community_bot_commands}>"
                         ),
                         colour=discord.Colour.red()
                     )
                 )
                 return
-            for repo_issue in message_repo_issue_map:
+
+            log.trace(f"Found {issues = }")
+            # Remove duplicates
+            issues = set(issues)
+
+            if len(issues) > MAXIMUM_ISSUES:
+                embed = discord.Embed(
+                    title=random.choice(ERROR_REPLIES),
+                    color=Colours.soft_red,
+                    description=f"Too many issues/PRs! (maximum of {MAXIMUM_ISSUES})"
+                )
+                await message.channel.send(embed=embed, delete_after=5)
+                return
+
+            for repo_issue in issues:
                 if not self.check_in_block(message, " ".join(repo_issue)):
-                    result = await self.fetch_issues({repo_issue[1]}, repo_issue[0], "python-discord")
-                    if isinstance(result, list):
-                        links.extend(result)
+                    result = await self.fetch_issues(repo_issue[1], repo_issue[0], "python-discord")
+                    if isinstance(result, IssueState):
+                        links.append(result)
```

Comment:
As the error message are now including in the same embed, we could possibly add them to the embed sent on `on_message` too.

Reply:
It is ignored on purpose, we shouldn't send automatic error embeds. It is part of the original specs of #566.

Reply:
Alright.